### PR TITLE
Rename HwcBuffer to HwcMeta

### PR DESCRIPTION
**Pull request by js0701**
HwcBuffer name is misleading, actually it is just some meta info

Tracked-On: None
Tests: None
Signed-off-by: Lin Johnson <johnson.lin@intel.com>

### CODE REVIEW

File: os/linux/platformdefines.h [hunk 1]
```diff
@@ -41,7 +41,7 @@ struct gbm_handle {
   } import_data;
   struct gbm_bo* bo = NULL;
   struct gbm_bo* imported_bo = NULL;
-  HwcBuffer meta_data_;
```
Comment (YuanjunHuang) [inline os/linux/platformdefines.h]: also change the file name?

File: public/hwcbuffer.h [hunk 2]
```diff
@@ -22,10 +22,10 @@
 
 #include <hwcdefs.h>
 
-struct HwcBuffer {
-  HwcBuffer() = default;
+struct HwcMeta {
+  HwcMeta() = default;
 
-  HwcBuffer &operator=(const HwcBuffer &rhs) = delete;
```
Comment (YuanjunHuang) [inline public/hwcbuffer.h]: also change the file name?

File: public/hwcmeta.h [hunk 3]
```diff
@@ -22,10 +22,10 @@
 
 #include <hwcdefs.h>
 
-struct HwcBuffer {
```
Comment (stripes416) [inline public/hwcmeta.h]: The #ifndef, #define, and #endif should be renamed to PUBLIC_HWCMETA_H_

**Comment by js0701** [comment]
@stripes416 Commit message updated

File: public/hwcmeta.h (hunk 3 above)
Comment (stripes416) [reply public/hwcmeta.h]: The #endif still needs to be updated. Easy to miss! 😃

**Comment by js0701** [comment]
@stripes416 Done. Appreciated the review

**Pull request merged**